### ISSUE TITLE
Clarification

### ISSUE THREAD

**Issue by anishacharya** [lightly-ai/lightly#1396]
https://github.com/lightly-ai/lightly/blob/100afe6dd82bf3ab362943a43d027530d24a11d5/lightly/loss/ntx_ent_loss.py#L165

Why is the final similarity matrix computed as:
01 | 00
10 | 11 

and not
00 | 01
10 | 11 

I am slightly confused.

**Comment by guarin** [comment]
Hi! It is computed like that to match the order of the `labels` tensor created some lines further down: https://github.com/lightly-ai/lightly/blob/100afe6dd82bf3ab362943a43d027530d24a11d5/lightly/loss/ntx_ent_loss.py#L170-L173

The labels tensor is a range repeated twice, so it looks like this: `[0, 1, 2, 3, 0, 1, 2, 3]`. This means that positive pairs in the logits array have to be in positions: `logits[0, 0], logits[1, 1], logits[2, 2], logits[3, 3,], logits[4, 0], logits[5, 1], logits[6, 2], logits[7, 3]`. As positive pairs should always be between different views of the same image, so from `logits_01` or `logits_10`, we put them in the first columns of the final tensor.

To make it more concrete, the logits tensor should have positive pairs in these positions:
```
1 0 0 0 | 0 0 0 0
0 1 0 0 | 0 0 0 0
0 0 1 0 | 0 0 0 0
0 0 0 1 | 0 0 0 0 
----------------
1 0 0 0 | 0 0 0 0
0 1 0 0 | 0 0 0 0
0 0 1 0 | 0 0 0 0 
0 0 0 1 | 0 0 0 0
``` 

And positive pairs are stored on the diagonals of logit_01 and logit_10.

**Comment by guarin** [comment]
Please reopen if you have more questions :)
